library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsChunkSource.java

```diff
@@ -358,7 +358,7 @@ public void getNextChunk(HlsMediaChunk previous, long playbackPositionUs, long l
             isTimestampMaster,
             timestampAdjuster,
             previous,
-            mediaPlaylist.drmInitData,
+            mediaPlaylist.drmInitData.get(segment.relativeDiscontinuitySequence),
```

Comment:
I think more than one drm init data can map to a single discontinuity sequence.

Reply:
Thank you for taking a look at this pr.

> more than one drm init data can map to a single discontinuity sequence

I coundn’t think of the situation of multiple init data in a single discontinuity sequence
Could you tell me how the playlist look like?
Do you mean we have to apply drmInitData until the next `#EXT-X-KEY` or the end of the playlist?

And one more thing.
As you explained [here](https://github.com/google/ExoPlayer/pull/4164#issuecomment-384239441), you don’t want to use SparseArray?
If so, how do you want me to implement it?